test/volume_test.py

```diff
@@ -367,3 +372,22 @@ async def test_open_files_error_annotation(tmp_path):
 def test_invalid_name(servicer, client, name):
     with pytest.raises(InvalidError, match="Invalid Volume name"):
         modal.Volume.lookup(name)
+
+
+def test_warn_on_local_volume_mount(client, servicer):
+    app = modal.App()
+    vol = modal.Volume.from_name("xyz")
+
+    assert modal.is_local() == True
+    with pytest.warns(match="local"):
+        app.function(volumes={"/foo": vol})(dummy).local()
```

Comment:
I'd suggest keeping the function decoration outside of the `pytest.warns` scope to make sure that we're warns when `.local()` is _called_, rather than at some point point.